### DISCUSSION TITLE
✨ [Feature]: 完善`Notification`元件功能

### DISCUSSION THREAD

**Discussion by fu050409** [DevCloudFE/vue-devui#1873] [Ideas]
看起来DevUI的全局通知元件的设计依旧不够完善。

首先是UI比较简陋，只有僵硬的一个弹窗，或许我们可以参考HyperUI中[`Alerts`](https://www.hyperui.dev/components/application-ui/alerts)的样式，或者添加一个`solid`和`outlined`的API来选择是否是填充色。

除此之外，还可以参考Element UI的[`Notification`](https://element.eleme.cn/#/zh-CN/component/notification)组件和PrimeVue的[`Toast`](https://primevue.org/toast/)元件，前者使用带颜色的图标来区分，后者则使用整体消息框的颜色和图标来区分通知类型。（个人更喜欢后一种，或许我们也可以提供选择接口，例如`filled`？）

其次是消息队列不够完善，经常出现多个通知重叠显示（Angular版本的DevUI样例中是完全重叠展示，~~这看起来也不够优雅~~），而且我们的消息队列是古板的按照顺序排列之后在原地消失，这意味着如果大量的通知堆叠在队列里，会有通知被埋藏在窗口最底部而**永远无法**被观察到，因为它出现在最底部，所以它不会上移。

我们是否应该对此做一些优化呢？

**Comment by GreatZPP** [comment]
> 其次是消息队列不够完善，经常出现多个通知重叠显示（Angular版本的DevUI样例中是完全重叠展示，这看起来也不够优雅），而且我们的消息队列是古板的按照顺序排列之后在原地消失，这意味着如果大量的通知堆叠在队列里，会有通知被埋藏在窗口最底部而永远无法被观察到，因为它出现在最底部，所以它不会上移。

此确实是不够好的，我们对于多个通知消失时的处理需要更优雅一些，需要加入自动上移的实现

> 首先是UI比较简陋，只有僵硬的一个弹窗，或许我们可以参考HyperUI中[Alerts](https://www.hyperui.dev/components/application-ui/alerts)的样式，或者添加一个solid和outlined的API来选择是否是填充色。
除此之外，还可以参考Element UI的[Notification](https://element.eleme.cn/#/zh-CN/component/notification)组件和PrimeVue的[Toast](https://primevue.org/toast/)元件，前者使用带颜色的图标来区分，后者则使用整体消息框的颜色和图标来区分通知类型。（个人更喜欢后一种，或许我们也可以提供选择接口，例如filled？）

此可能涉及到咱们设计体系与风格，我们后续可以从整体组件层面展开来看下
